src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -74,8 +74,16 @@ public SonarWebService(IDownloader downloader, string server, ILogger logger)
 
                 var json = JObject.Parse(contents);
                 var profiles = json["profiles"].Children<JObject>();
-
-                var profile = profiles.SingleOrDefault(p => language.Equals(p["language"].ToString()));
+                JObject profile = null;
+                try
+                {
+                    profile = profiles.SingleOrDefault(p => language.Equals(p["language"].ToString()));
+                }
+                catch (InvalidOperationException) //As we don't have fail-fast policy for unsupported version for now, we should handle gracefully multi-QPs set for a project, here for SQ < 6.7
+                {
+                    logger.LogWarning($"Your project has multiple Quality Profiles defined for language `{language}`, taking the default one.");
```

Comment:
This message is incorrect and misleading; a user reading this will go to SonarQube to look at their project configuration to work out how they have multiple QPs defined, which is impossible, isn't it?

The real problem is that the user is using an old version of SQ that is now unsupported, so the message should say that.

Reply:
I just rebased the branch now, i'll modify the message indeed.